datastore/gcloud/aio/datastore/datastore.py

```diff
@@ -43,13 +52,21 @@ def __init__(self, project: Optional[str] = None,
         self.namespace = namespace
 
         self.session = session
-        self.token = token or Token(service_file=service_file, session=session,
-                                    scopes=SCOPES)
+
+        if _IS_DEV:
+            # Tokens are unneeded when using dev emulator.
+            self.token = None
+        else:
+            self.token = token or Token(service_file=service_file, session=session,
+                                        scopes=SCOPES)
 
     async def project(self) -> str:
         if self._project:
             return self._project
 
+        if _IS_DEV:
```

Comment:
I _think_ (but could be convinced either way) that this should take precedence over any `_project` value we've explicitly set, given we're effectively hijacking other user args such as `service_account` when an emulator is detected. Thoughts?

Reply:
Worth consideration...
I think in general if a developer is going to be changing the project name between dev and prod (or staging) they have to be obtaining that project id from some variable. Either passed in to the running program or pulled from an env var. So I largely would think it wouldn't really matter.
I guess the question is if there are cases where someone would be connecting to 2 different projects in their code. And if so the emulator supports that or not... I can go either way.

Reply:
> 2 different projects
That's a great thought. We do that in a bunch of our codebases, to communicate between, say, the engineering production env and the datascience one. I suppose that would just be unsupported by the emulator. I can't really think of a decent way to support that sort of cross-project work involving an emulator, though, so I might put that off for a later version only if need be.